### ISSUE TITLE
Socket#connect_nonblock working?

### ISSUE THREAD

**Issue by ioquatix** [oracle/truffleruby#1461]
I print out the status from

```
  def connect_nonblock(sockaddr, exception: true)
    fcntl(Fcntl::F_SETFL, Fcntl::O_NONBLOCK)

    if sockaddr.is_a?(Addrinfo)
      sockaddr = sockaddr.to_sockaddr
    end

    status = Truffle::Socket::Foreign.connect(descriptor, sockaddr)

    if status < 0
      if exception
        Truffle::Socket::Error.write_nonblock('connect(2)')
      else
        :wait_writable
      end
    else
      0
    end
  end
```

and it's always -1.

The exact same code works fine on MRI.

**Comment by ioquatix** [comment]
What's odd is that the server accepts the connection but it continues to return -1:

```
RSpec is shutting down and will print the summary report... Interrupt again to force quit.
Server accept...
Client connect...
status=-1
Server read...
status=-1
status=-1
status=-1
status=-1
status=-1
status=-1
```

**Comment by ioquatix** [comment]
Okay so I found out the problem. You are not correctly handling `Errno::EISCONN`.

**Comment by ioquatix** [comment]
Here is a monkey patch: https://github.com/socketry/async-io/commit/3380a7dfe4e973b7abf383bd3f7508ccc14f906f

**Comment by eregon** [comment]
Fixed by #1465.

**Comment by ioquatix** [comment]
Thanks!
